compiler/src/dotty/tools/dotc/config/CliCommand.scala

```diff
@@ -191,3 +144,56 @@ trait CliCommand:
 
   extension [T](setting: Setting[T])
     protected def value(using ss: SettingsState): T = setting.valueIn(ss)
+
+  extension (s: String)
+    def padLeft(width: Int): String = StringBuilder().tap(_.append(" " * (width - s.length)).append(s)).toString
+
+  // Formatting for -help and -Vphases in two columns, handling long field1 and wrapping long field2
+  class Columnator(heading1: String, heading2: String, maxField: Int, separation: Int = 2):
+    def apply(texts: List[List[(String, String)]])(using Context): String = StringBuilder().tap(columnate(_, texts)).toString
+
+    private def columnate(sb: StringBuilder, texts: List[List[(String, String)]])(using Context): Unit =
+      import scala.util.Properties.{lineSeparator => EOL}
+      import Highlighting.*
+      val colors = Seq(Green(_), Yellow(_), Magenta(_), Cyan(_), Red(_))
+      val nocolor = texts.length == 1
+      def color(index: Int): String => Highlight = if nocolor then NoColor(_) else colors(index % colors.length)
+      val maxCol = ctx.settings.pageWidth.value
+      val field1 = maxField.min(texts.flatten.map(_._1.length).filter(_ < maxField).max) // widest field under maxField
+      val field2 = if field1 + separation + maxField < maxCol then maxCol - field1 - separation else 0 // skinny window -> terminal wrap
+      val separator = " " * separation
+      def formatField1(text: String): String = if text.length <= field1 then text.padLeft(field1) else text + EOL + "".padLeft(field1)
```

Comment:
```suggestion
      def formatField1(text: String): String = if text.length <= field1 then text.padLeft(field1) else text + "\n" + "".padLeft(field1)
```

We had some issues with `EOL` and Windows in the past. The `"\n"` seems to be what we should use by default.

Reply:
I have followed the issue, but usually if I'm on cygwin I want the correct line separator. I'll try to come up with a use case slash test.

Reply:
> I'll try to come up with a use case slash test.
Maybe if I start using cygwin again on a regular basis...